redbot/cogs/permissions/resolvers.py

```diff
@@ -12,20 +12,10 @@
     Returns the value from a check if it is valid
     """
 
-    # Non staticmethods should not be run without their parent
-    # class, even if the parent class did not deregister them
-    if check.__module__ is None:
-        pass
-    elif isinstance(check, types.FunctionType):
-        if (
-            next(filter(lambda x: check.__module__ == x.__module__, ctx.bot.cogs.values()), None)
-            is None
-        ):
-            return None
-
     val = None
     # let's not spam the console with improperly made 3rd party checks
     try:
+        # pretty sure this is redunant, but if it is, it will short circuit at or
         if asyncio.iscoroutine(check) or asyncio.iscoroutinefunction(check):
```

Comment:
This should just be `iscoroutinefunction()`.

`iscoroutine()` checks if an object is a coroutine object.